sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java

```diff
@@ -0,0 +1,165 @@
+ */
+
+package org.sormas.e2etests.steps.web.application.entries;
+
+import static org.sormas.e2etests.pages.application.entries.CreateNewTravelEntryPage.*;
+import static org.sormas.e2etests.pages.application.entries.EditTravelEntryPage.*;
+
+import cucumber.api.java8.En;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import org.sormas.e2etests.entities.pojo.helpers.ComparisonHelper;
+import org.sormas.e2etests.entities.pojo.web.TravelEntry;
+import org.sormas.e2etests.entities.services.TravelEntryService;
+import org.sormas.e2etests.helpers.WebDriverHelpers;
+import org.sormas.e2etests.state.ApiState;
+
+public class CreateNewTravelEntrySteps implements En {
+  private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+  private final WebDriverHelpers webDriverHelpers;
+  public static TravelEntry travelEntry;
+  public static TravelEntry aTravelEntry;
+
+  @Inject
+  public CreateNewTravelEntrySteps(
+      WebDriverHelpers webDriverHelpers, TravelEntryService travelEntryService, ApiState apiState) {
+    this.webDriverHelpers = webDriverHelpers;
+
+    When(
+        "^I fill the required fields in a new travel entry form$",
+        () -> {
+          travelEntry = travelEntryService.buildGeneratedEntry();
+          fillFirstName(travelEntry.getFirstName());
+          fillLastName(travelEntry.getLastName());
+          selectSex(travelEntry.getSex());
+          selectResponsibleRegion(travelEntry.getResponsibleRegion());
+          selectResponsibleDistrict(travelEntry.getResponsibleDistrict());
+          selectResponsibleCommunity(travelEntry.getResponsibleCommunity());
+          fillDisease(travelEntry.getDisease());
+          fillPointOfEntry(travelEntry.getPointOfEntry());
+          fillPointOfEntryDetails(travelEntry.getPointOfEntryDetails());
+        });
+
+    When(
+        "^I fill the required fields in a new case travel entry form$",
+        () -> {
+          travelEntry = travelEntryService.buildGeneratedEntry();
+          selectResponsibleRegion(travelEntry.getResponsibleRegion());
+          selectResponsibleDistrict(travelEntry.getResponsibleDistrict());
+          selectResponsibleCommunity(travelEntry.getResponsibleCommunity());
+          fillPointOfEntry(travelEntry.getPointOfEntry());
+          fillPointOfEntryDetails(travelEntry.getPointOfEntryDetails());
+        });
+
+    When(
+        "^I save the travel entry$",
+        () -> {
+          webDriverHelpers.clickOnWebElementBySelector(SAVE_BUTTON);
+        });
+
+    When(
+        "^I navigate to person tab in Edit travel entry page$",
+        () -> {
+          webDriverHelpers.clickOnWebElementBySelector(TRAVEL_ENTRY_PERSON_TAB);
+        });
+
+    When(
+        "I check the created data is correctly displayed on Edit travel entry page for DE version",
+        () -> {
+          TimeUnit.SECONDS.sleep(2);
```

Comment:
why are we adding this sleep?

Reply:
We're adding it to ensure that DOM finishes reloading before we start to gather data from the page.
Without it the test would tend to throw a stale reference error, seemingly caused by the data collection method kicking in too quickly and latching onto the old DOM from previous step's popup.